core/consensus/timeline/types.hpp

```diff
@@ -26,6 +26,54 @@ namespace kagome::consensus {
   /// slot number of the block production
   using SlotNumber = uint64_t;
 
+  /// duration of single slot in milliseconds
+  struct SlotDuration : public std::chrono::milliseconds {
+    SlotDuration() = default;
+
+    template <typename Rep, typename Period>
+    SlotDuration(const std::chrono::duration<Rep, Period> &duration)
+        : std::chrono::milliseconds(
+            std::chrono::duration_cast<std::chrono::milliseconds>(duration)
+                .count()) {}
+
+    template <class Rep>
+      requires std::is_integral_v<Rep>
+    SlotDuration(Rep duration) : std::chrono::milliseconds(duration) {}
```

Comment:
Do you want to avoid explicit `duration_cast`?
```suggestion
    using std::chrono::seconds::milliseconds;
    SlotDuration(std::chrono::milliseconds s) : std::chrono::milliseconds{s} {}
```

Reply:
It breaks some operator. I left it as is